qiskit/transpiler/passes/unroller.py

```diff
@@ -68,9 +70,13 @@ def run(self, dag):
                 raise QiskitError('Error decomposing node {}: {}'.format(node.name, err))
 
             if not rule:
-                raise QiskitError("Cannot unroll the circuit to the given basis, %s. "
-                                  "No rule to expand instruction %s." %
-                                  (str(self.basis), node.op.name))
+                if 'unitary' in self.basis:
+                    rule = [(UnitaryGate(Operator(node.op), label=node.name),
```

Comment:
Is `UnitaryGate(node.op)` sufficient?

Reply:
The init of `Operator` handles a few more types of inputs than `UnitaryGate` so I used that to be safe.